### DISCUSSION TITLE
How to calculate Median time-to-event based on only the subset of patients who actually experienced an event (rather than of the total number of patients)

### DISCUSSION THREAD

**Discussion by chrispoptic** [CamDavidsonPilon/lifelines#1372] [Q&A]
How to calculate Median time-to-event based on the total number of patients who actually experienced an event (rather than of the total number of patients)?

The `median_survival_time_` attribute of the `KaplanMeierFitter` class returns the estimated median time to event. 

But this median survival time is evaluated for all patients, regardless of whether the experienced an event or not.
For Kaplan Meier curves where survivor probability never drops below 50%, this leads to returning `numpy.inf`.

Instead, how to calculate the `median_survival_time_` for only the subset of patients who _actually_ experienced an event?

**Comment by CamDavidsonPilon** [comment]
Hi @chrispoptic, 

I suppose you would just fit the KaplanMeier to only the patients that experienced the event, then call `median_survival_time_`.
